index.d.ts

```diff
@@ -524,7 +524,11 @@ interface OptionProps extends TableRowOwnProps {
 }
 
 export interface AutocompleteItemProps extends OptionProps {
-  children?: JSX.Element | null
+  children?: JSX.Element | null,
+  /**
+   * The item of the autocomplete items
+   */
+   item: any,
```

Comment:
Have you considered making item a generic to ensure type safety? I think TypeScript is relatively good at inferring the value without explicitly having the pass the generic type.

Reply:
I see below that we're using propTypes. Not sure if this has any implications, but it could explain this decision.